table/block_based/block_based_table_builder.cc

```diff
@@ -461,6 +465,10 @@ struct BlockBasedTableBuilder::Rep {
       buffer_limit = std::min(tbo.target_file_size,
                               compression_opts.max_dict_buffer_bytes);
     }
+    // TODO figure out c++ member initialization syntax
+    buffer_exceeds_global_memory_limit = false;
+    cache_rep = (table_options.block_cache != nullptr ? new CacheRep(tbo.block_cache.get()) : nullptr);
```

Comment:
Question 1:
While the block cache in the WriteBufferManager's case is passed in on WBM's creation, for BlockBasedTableBuilder, I believe we are using the block cache passed in the table_options. Am I correct?